normandy/recipes/tests/test_commands.py

```diff
@@ -257,3 +258,42 @@ def test_it_does_not_resign_up_to_date_actions(self, settings, mocked_autograph)
         call_command("update_action_signatures")
         a.refresh_from_db()
         assert a.signature.signature == "original signature"
+
+
+@pytest.mark.django_db
+class TestUpdateAddonUrls(object):
+    def test_it_works(self):
+        action = ActionFactory(name="opt-out-study")
+        recipe = RecipeFactory(
+            action=action,
+            arguments={"addonUrl": "https://before.example.com/extensions/addon.xpi"},
+        )
```

Comment:
Perhaps it'd be worth adding something like:
```python
    other_action = ActionFactory(name="anything-else")
    other_recipe = RecipeFactory(
        action=other_action,
        arguments={"addonUrl": "https://before.example.com/extensions/addon.xpi"},
    )
    call_command("update_addon_urls", "after.example.com")
    other_recipe.refresh_from_db()
    # Should *not* change.
    assert other_recipe.arguments["addonUrl"] == "https://before.example.com/extensions/addon.xpi"
```

Reply:
:+1:

Reply:
Wait, what?! What "real problem".

Reply:
The real problem was that I trying to update the revision arguments directly, like`revision.arguments["updateUrl"] = ...`. That made nothing work at all, and was very confusing, so I was trying random things.